### ISSUE TITLE
Image morphology convergence does not work anymore

### ISSUE THREAD

**Issue by dmi3kno** [ropensci/magick#249]
I did not notice when this stopped working, but image convergence (using `-1` instead of number of iterations) now throws an error.
![](http://www.imagemagick.org/Usage/morphology/man.gif)
```r
library(magick)
image_read("http://www.imagemagick.org/Usage/morphology/man.gif") %>% 
  image_morphology("Thinning", "Skeleton", -1)
#> Error in magick_image_write(image, format, quality, depth, density, comment): R: negative or zero image size `' @ error/image.c/CloneImage/819
```
<sup>Created on 2020-05-11 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

And it should produce 
![](http://www.imagemagick.org/Usage/morphology/man_raw_thinned.gif)

Example from [here](http://www.imagemagick.org/Usage/morphology/#thinning_skeleton)

**Comment by jeroen** [comment]
Is this a change in `magick` or in `imagemagick`? Which OS are you on again?

**Comment by dmi3kno** [comment]
I am on Ubuntu, but that shouldn't matter.
<details closed>
<summary> <span title='Click to Expand'> Session Info </span> </summary>

```r

─ Session info ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
 setting  value                       
 version  R version 3.6.3 (2020-02-29)
 os       Ubuntu 18.04.4 LTS          
 system   x86_64, linux-gnu           
 ui       RStudio                     
 language en_GB:en                    
 collate  en_GB.UTF-8                 
 ctype    en_GB.UTF-8                 
 tz       Europe/Oslo                 
 date     2020-05-12                  

─ Packages ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
 package     * version  date       lib source                          
 assertthat    0.2.1    2019-03-21 [1] CRAN (R 3.6.0)                  
 backports     1.1.6    2020-04-05 [1] CRAN (R 3.6.3)                  
 callr         3.4.3    2020-03-28 [1] CRAN (R 3.6.3)                  
 cli           2.0.2    2020-02-28 [1] CRAN (R 3.6.3)                  
 crayon        1.3.4    2017-09-16 [1] CRAN (R 3.6.0)                  
 curl          4.3      2019-12-02 [1] CRAN (R 3.6.1)                  
 fansi         0.4.1    2020-01-08 [1] CRAN (R 3.6.2)                  
 glue          1.4.0    2020-04-03 [1] CRAN (R 3.6.3)                  
 magick      * 2.3.9000 2020-05-12 [1] Github (ropensci/magick@df053a8)
 magrittr      1.5      2014-11-22 [1] CRAN (R 3.6.3)                  
 pkgbuild      1.0.6    2019-10-09 [1] CRAN (R 3.6.1)                  
 prettyunits   1.1.1    2020-01-24 [1] CRAN (R 3.6.3)                  
 processx      3.4.2    2020-02-09 [1] CRAN (R 3.6.3)                  
 ps            1.3.2    2020-02-13 [1] CRAN (R 3.6.3)                  
 R6            2.4.1    2019-11-12 [1] CRAN (R 3.6.1)                  
 Rcpp          1.0.4.6  2020-04-09 [1] CRAN (R 3.6.3)                  
 remotes       2.1.0    2019-06-24 [1] CRAN (R 3.6.0)                  
 rprojroot     1.3-2    2018-01-03 [1] CRAN (R 3.6.0)                  
 rstudioapi    0.11     2020-02-07 [1] CRAN (R 3.6.3)                  
 sessioninfo   1.1.1    2018-11-05 [1] CRAN (R 3.6.0)                  
 withr         2.2.0    2020-04-20 [1] CRAN (R 3.6.3)                  

[1] /home/dm0737pe/R/x86_64-pc-linux-gnu-library/3.6
[2] /usr/local/lib/R/site-library
[3] /usr/lib/R/site-library
[4] /usr/lib/R/library

```

</details>
<br>

**Comment by jeroen** [comment]
Did you build imagemagick from source? That's not the version of imagemagick that is included with ubuntu.

Did this work before? I can't think of anything I have changed that could have broken this, it is more likely a change between imagemagick 6 and 7 ?

**Comment by jeroen** [comment]
So it looks like the imagemagick call to image_morphology with `iter = -1` now creates an output image without a width and a height, very strange.

**Comment by dmi3kno** [comment]
I re-built imagemagick from source (version 7.0.10.11) and it works in commandline:
```bash
convert man.gif -morphology Thinning:-1 Skeleton  man_raw_skeleton.gif
```
Producing skeleton as expected. 

I can also confirm that even with the new ImageMagick, the image produced by `image_morphology(...,iter=-1)` is of size  `0x0`, so something must be happening in the wrapper to Magick++ library. Let me know if you want me to test anything

**Comment by dmi3kno** [comment]
Any news on Skeletons in the Magick closet?

**Comment by dmi3kno** [comment]
May I ping this issue one more time? Any updates @jeroen ?

**Comment by jeroen** [comment]
I have to ask @dlemstra about this but I haven't had time to create a small c++ example.

**Comment by jeroen** [comment]
Related: https://github.com/ropensci/magick/commit/8089c1ec70e74b41e6b4bbb50ed512ab0c1015dd

**Comment by jeroen** [comment]
OK I reverted https://github.com/ropensci/magick/commit/8089c1ec70e74b41e6b4bbb50ed512ab0c1015dd. I had to make that change for CRAN because it was causing undefined behaviour when using negative iterations, but I don't remember the details. I'll just re-enable it.

**Comment by dmi3kno** [comment]
I don't know if you are aware, there's only one legit negative value and that it "-1", which means[ "repeat until converging"](https://legacy.imagemagick.org/Usage/morphology/#verbose).

**Comment by dmi3kno** [comment]
Also confirming this works. Thank you very much!

**Comment by jeroen** [comment]
This fix is on CRAN now in magick 2.5.1
